File: Muc_Do_3_Kha_Kho/Bai14_CapSoCong/Solution.cpp
#include <bits/stdc++.h>
using namespace std;
// Bai 14: Cap so cong
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    if (fopen("BAI14.INP", "r")) {
        freopen("BAI14.INP", "r", stdin);
        freopen("BAI14.OUT", "w", stdout);
    }
    int n, q;
    if (!(cin >> n >> q)) return 0;
    vector<long long> diff1(n + 2, 0);
    vector<long long> diff2(n + 2, 0);
    while (q--) {
        int l, r;
        long long d;
        cin >> l >> r >> d;
        diff2[l] += d;
        diff2[r + 1] -= d;
        diff1[r + 1] -= (r - l + 1) * d;
    }
    for (int i = 1; i <= n; ++i) {
        diff1[i] += diff1[i-1] + diff2[i];
    }
    long long cur = 0;
    for (int i = 1; i <= n; ++i) {
        cur += diff1[i];
        cout << cur << (i == n ? "" : " ");
    }
    cout << "\n";
    return 0;
}

File: Muc_Do_3_Kha_Kho/Bai14_CapSoCong/Solution.py
import sys
def solve():
    try:
        sys.stdin = open('BAI14.INP', 'r')
        sys.stdout = open('BAI14.OUT', 'w')
    except:
        pass
    data = sys.stdin.read().split()
    if not data: return
    n = int(data[0])
    q = int(data[1])
    idx = 2
    diff1 = [0] * (n + 2)
    diff2 = [0] * (n + 2)
    for _ in range(q):
        l = int(data[idx])
        r = int(data[idx+1])
        d = int(data[idx+2])
        idx += 3
        diff2[l] += d
        diff2[r+1] -= d
        diff1[r+1] -= (r - l + 1) * d
        
    for i in range(1, n + 1):
        diff1[i] += diff1[i-1] + diff2[i]
        
    cur = 0
    ans = []
    for i in range(1, n + 1):
        cur += diff1[i]
        ans.append(str(cur))
    print(' '.join(ans))
solve()
